### ISSUE TITLE
Optionally allow adding `- returns:` docs to initializers

### ISSUE THREAD

**Issue by mohpor** [realm/SwiftLint#557]
Hi,
I use [VVDocumenter](https://github.com/onevcat/VVDocumenter-Xcode) to help me document my codes.
`init()` functions return an objects, although not explicitly defined in their syntax. For this particular reason, SwiftLint-ing them complains that their documentations are invalid.

example:

``` swift
 /**
   Generates the view programatically to avoid giving out informations about the alert's views.   
   - parameter frame: A frame. Will be ignored.   
   - returns: An instance of UIView.
   */
  override init(frame: CGRect) {...}

```

(Disclaimer: This is just an example)

Swift lint will complain and say:

>  warning: Valid Docs Violation: Documented declarations should be valid. (valid_docs)

Although it is perfectly valid. You might argue that we don't need to document the return part, but I have many cases that it helps clarify the resulted instance, especially in case of failable initializers.

**UPDATE**:
Just to be clear, removing `- returns` from documentation does actually solve the validation problem.

**Comment by mohpor** [comment]
I'm more than happy to submit a PR but I jus don't know how to check if the `kind` is a constructor.

```swift
func delcarationReturns(declaration: String, kind: SwiftDeclarationKind? = nil) -> Bool {
    if let kind = kind where SwiftDeclarationKind.variableKinds().contains(kind) {
        return true
    }

    guard let outsideBracesMatch = matchOutsideBraces(declaration) else {
        return false
    }

    return outsideBracesMatch.containsString("->")
}

```

in `ValidDocsRule.swift` should return `true` in case `kind` is `FunctionConstructor` but no matter what, it never detects `init` functions as one!

To sum it up I felt pretty stupid trying to contribute to this repo. I couldn't find a way to extract function name from SourceKitten and .... (sigh)

**Comment by startupthekid** [comment]
There's another thing it misses too:

```swift
func myFunction(closure: (firstInt: Int, secondInt: Int) throws -> Void) { 
}
```

If a closure argument can throw an error, `valid_docs` complains because `- throws:` isn't in the documentation. In this example though, the function itself doesn't through, the closure parameter does.

**Comment by chadmoone** [comment]
Dupe of #439?

**Comment by Grubas7** [comment]
I have same issue.

**Comment by jpsim** [comment]
this is a good "starter bug" if anyone is interested in contributing to SwiftLint.

**Comment by mohpor** [comment]
@startupthekid 
> If a closure argument can throw an error, valid_docs complains because - throws: isn't in the documentation. In this example though, the function itself doesn't through, the closure parameter does.

It seems like a separate issue.
Care to open a new issue?

**Comment by jpsim** [comment]
Fixed in #739.
